docs/orchestrating-elastic-stack-applications/maps.asciidoc

```diff
@@ -0,0 +1,211 @@
+
+The {ems} configuration file is automatically setup by ECK to establish a secure connection to Elasticsearch.
+[id="{p}-maps-data"]
+== Map data
+The {ems} Docker image contains only a few zoom levels of data. In order to get the map data up to the highest zoom level {ems} needs a basemap file mounted into its container.
+[id="{p}-maps-basemap-download"]
+=== Basemap download
+You have to download the basemap ahead of time on a machine that is not air-gapped and populate a volume that can be mounted into the {ems} Pods.  See also the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server[{ems} documentation.]
+
+The exact procedure of how to get a Kubernetes volume populated with that data is out of scope of this documenation as it depends on your specific Kubernetes setup and choice of volume provisioner. One possible approach that should work for most setups is as follows:
+
+. Download the basemap zip archive using the link shown in the {ems} UI or extracted from the `/status` endpoint.
+. Create a PersistentVolumeClaim of sufficient size (> 90G) and a temporary Pod to mount the corresponding volume
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ems-basemap
+spec:
+  storageClassName: "standard"
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 250G
+---
+kind: Pod
+apiVersion: v1
+metadata:
+  name: ems-data-setup
+spec:
+  terminationGracePeriodSeconds: 0
+  volumes:
+    - name: ems-storage
+      persistentVolumeClaim:
+       claimName: ems-basemap
+  containers:
+    - name: ems-setup
+      image: ubuntu
+      command: [bash, -c, "apt-get update && apt-get install unzip && while true; do sleep 10; done"]
+      volumeMounts:
+        - mountPath: "/usr/share/planet"
+          name: ems-storage
+----
+. Use `kubectl` to copy the basemap data into the volume
++
+[source,sh]
+----
+kubectl cp planet.zip ems-data-setup:/usr/share/planet/planet.zip
+----
+. Unzip the archive on the temporary Pod
++
+[source,sh]
+----
+kubectl exec ems-data-setup -- unzip /usr/share/data/planet.zip -d /usr/share/planet
+----
+. Delete the temporary Pod and remount the volume into the {ems} Pods as shown <<{p}-maps-pod-configuration,below>>.
++
+[source,sh]
+----
+kubectl delete pod ems-data-setup
+----
+
+[id="{p}-maps-pod-configuration"]
+=== Pod configuration
+You can <<{p}-customize-pods,customize the {ems} Pod>> using a Pod template.
+
+The following example demonstrates how to create a {ems} deployment which mounts a data volume with the complete basemap.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: ems-sample
```

Comment:
Maybe keep the same name in the example, when the previous steps have been followed it helps the user to proceed with `kubectl apply -f` :
```suggestion
  name: quickstart
```